src/JavaCodeGenerator.cs

```diff
@@ -250,2859 +283,3299 @@ private static CodeModel TransformCodeModel(CodeModel codeModel, JavaSettings se
+                            methodImports.AddRange(CompositeTypeImports(typeName, false, false, true, false, settings.Package));
+                        }
+                        else if (methodIsPagingNonPollingOperation)
+                        {
+                            methodImports.AddRange(CompositeTypeImports(typeName, false, false, true, false, settings.Package));
                         }
                     }
+                    serviceClientImports.AddRange(methodImports);
                 }
             }
-            return xmlSequenceWrappers;
+
+            return new ServiceClient(serviceClientClassName, serviceClientInterfaceName, serviceClientImports, serviceClientRestAPI, serviceClientMethodGroupClients, serviceClientProperties, serviceClientConstructors, serviceClientMethods);
         }
 
-        private static IEnumerable<ServiceModel> ParseModels(CodeModel codeModel, JavaSettings settings)
+        private static MethodGroupClient ParseMethodGroupClient(AutoRestMethodGroup methodGroup, JavaSettings settings)
         {
-            List<ServiceModel> models = new List<ServiceModel>();
-            ServiceModels serviceModels = new ServiceModels();
-            foreach (CompositeType modelType in GetModelTypes(codeModel, settings))
+            string interfaceName = methodGroup.Name.ToString().ToPascalCase();
+            if (!interfaceName.EndsWith('s'))
             {
-                models.Add(ParseModel(modelType, settings, serviceModels));
+                interfaceName += 's';
             }
-            return models;
+
+            string className = interfaceName + (settings.IsFluent ? "Inner" : "Impl");
+
+            string methodGroupClientInterfacePath = settings.Package + "." + interfaceName;
+
+            string restAPIName = methodGroup.Name.ToString().ToPascalCase();
+            if (!restAPIName.EndsWith('s'))
+            {
+                restAPIName += 's';
+            }
+            restAPIName += "Service";
+            string restAPIBaseURL = methodGroup.CodeModel.BaseUrl;
+            List<RestAPIMethod> restAPIMethods = new List<RestAPIMethod>();
+            foreach (AutoRestMethod method in methodGroup.Methods)
+            {
+                restAPIMethods.Add(ParseRestAPIMethod(method, settings));
+            }
+            RestAPI restAPI = new RestAPI(restAPIName, restAPIBaseURL, restAPIMethods);
+
+            List<string> implementedInterfaces = new List<string>();
+            if (!settings.IsFluent)
+            {
+                implementedInterfaces.Add(interfaceName);
+            }
+
+            string serviceClientName = methodGroup.CodeModel.Name + "Impl";
+
+            string variableType = interfaceName + (settings.IsFluent ? "Inner" : "");
+            string variableName = interfaceName.ToCamelCase();
+
+            IEnumerable<ClientMethod> clientMethods = ParseClientMethods(restAPI, settings);
+
+            return new 
```

Comment:
Would just encourage in general labeling non-obvious null and boolean arguments--and in some cases where there are large numbers of boolean parameters consider adding overloads or an enum based on actual usage.